### PR TITLE
Add python-resolvelib package to support ansible-galaxy

### PR DESCRIPTION
**Pull request by sameluch**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
The ansible spec file within Mariner is missing a dependency on `python-resolvelib` for the ansible-galaxy command set. This pr adds and builds this python package within Mariner.

###### Change Log  

- Add python-resolvelib package to Mariner
- Add runtime requires on python-resolvelib to ansible spec 

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local build
- install ansible rpm in mariner container with resolvelib package
- check failing command `ansible-galaxy --help` post install for errors

### CODE REVIEW

**Comment by anphel31** [comment]
LGTM, but can you please run x64 + arm64 buddybuilds for the new package

**Review by anphel31 — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

File: SPECS/ansible/ansible.spec [hunk 1]
```diff
@@ -46,6 +47,9 @@ python3 setup.py test
 %{python3_sitelib}/*
 
 %changelog
+* Mon Jun 26 2023 Sam Meluch <sammeluch@microsoft.com> - 2.12.1-3
+- add python-resolvelib dependency to fix ansible-galaxy
+
 * Mon Jan 31 2022 Thomas Crain <thcrain@microsoft.com> - 2.12.1-2
```
Comment (oliviacrain) [inline SPECS/ansible/ansible.spec:53]: Small request:

```suggestion
* Mon Jan 31 2022 Olivia Crain <oliviacrain@microsoft.com> - 2.12.1-2
```

File: SPECS/python-resolvelib/python-resolvelib.spec [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+%global pypi_name resolvelib
+Summary:        Resolve abstract dependencies into concrete ones
+Name:           python-%{pypi_name}
+Version:        1.0.1
+Release:        1%{?dist}
+License:        ISC
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/sarugaku/resolvelib
+Source0:        %{pypi_source}#/python-resolvelib-1.0.1.tar.gz
```
Comment (oliviacrain) [inline SPECS/python-resolvelib/python-resolvelib.spec:11]:
```suggestion
Source0:        %{pypi_source}#/%{name}-%{version}.tar.gz
```

**Review by oliviacrain — APPROVED** [review]

File: SPECS/ansible/ansible.spec (hunk 1 above)
Comment (sameluch) [reply SPECS/ansible/ansible.spec]: Updated!

**Comment by sameluch** [comment]
> LGTM, but can you please run x64 + arm64 buddybuilds for the new package

Started a buddy build for `python-resolvelib` and `ansible` here: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=392184&view=results

**Pull request merged**